File: algorithms/06_javascript/sorting/quick_sort.js
function quickSort(arr) {
  if (arr.length <= 1) return arr;
  const pivot = arr[arr.length >> 1];
  const left = arr.filter(x => x < pivot);
  const mid  = arr.filter(x => x === pivot);
  const right= arr.filter(x => x > pivot);
  return [...quickSort(left), ...mid, ...quickSort(right)];
}
module.exports = { quickSort };

File: algorithms/08_r/sorting/quick_sort.R
quick_sort <- function(arr) {
  if (length(arr) <= 1) return(arr)
  pivot <- arr[ceiling(length(arr) / 2)]
  left <- arr[arr < pivot]
  mid <- arr[arr == pivot]
  right <- arr[arr > pivot]
  return(c(quick_sort(left), mid, quick_sort(right)))
}
